ocp_resources/data_import_cron.py

```diff
@@ -1,97 +1,85 @@
-        self.bind_immediate_annotation = bind_immediate_annotation
+        self.managed_data_source = managed_data_source
+        self.retention_policy = retention_policy
+        self.schedule = schedule
+        self.template = template
 
     def to_dict(self) -> None:
         super().to_dict()
+
         if not self.kind_dict and not self.yaml_file:
-            if self.image_stream and self.url:
-                raise ValueError("imageStream and url cannot coexist")
-
-            if not self.pull_method:
-                raise MissingRequiredArgumentError(argument="pull_method")
-
-            self.res.update({
-                "spec": {
-                    "template": {"spec": {"source": {"registry": {"pullMethod": self.pull_method}}}},
-                }
-            })
-            spec = self.res["spec"]["template"]["spec"]
-
-            if self.bind_immediate_annotation:
-                self.res["metadata"].setdefault("annotations", {}).update({
-                    f"{NamespacedResource.ApiGroup.CDI_KUBEVIRT_IO}/storage.bind.immediate.requested": ("true")
-                })
-            if self.image_stream:
-                spec["source"]["registry"]["imageStream"] = self.image_stream
-            if self.url:
-                spec["source"]["registry"]["url"] = self.url
-            if self.cert_configmap:
-                spec["source"]["registry"]["certConfigMap"] = self.cert_configmap
-            if self.schedule:
-                self.res["spec"]["schedule"] = self.schedule
+            if self.managed_data_source is None:
+                raise MissingRequiredArgumentError(argument="self.managed_data_source")
+
+            if self.schedule is None:
+                raise MissingRequiredArgumentError(argument="self.schedule")
+
+            if self.template is None:
+                raise MissingRequiredArgumentError(argument="self.template")
+
+            self.res["spec"] = {}
+            _spec = self.res["spec"]
+
+            _spec["managedDataSource"] = self.managed_data_source
+            _spec["schedule"] = self.schedule
+            _spec["template"] = self.template
+
             if self.garbage_collect:
-                self.res["spec"]["garbageCollect"] = self.garbage_collect
-            if self.managed_data_source:
-                self.res["spec"]["managedDataSource"] = self.managed_data_source
+                _spec["garbageCollect"] = self.garbage_collect
+
             if self.imports_to_keep:
-                self.res["spec"]["importsToKeep"] = self.imports_to_keep
-
-            storage = {}
-            if self.size:
-                storage["resources"] = {"requests": {"storage": self.size}}
-            if self.storage_class:
-                storage["storageClassName"] = self.storage_class
-            if storage:
-                spec["storage"] = storage
+                _spec["importsToKeep"] = self.imports_to_keep
+
+            if self.retention_policy:
```

Comment:
@geetikakay did you added code manually or this is what the generator generate for you? 
(I think the generetor should do `if not None`)

Reply:
@myakove class-generator generated this code.

Reply:
Do you mind regenerate after this one is merged https://github.com/RedHatQE/openshift-python-wrapper/pull/2281 ?

Reply:
Done